### PR TITLE
test: Enhanced running in unstable environments

### PR DESCRIPTION
**Pull request by utam0k**
## Description

Fix [a failed of this job](https://werft.gitpod-dev.com/job/gitpod-custom-main.185)

## Related Issue(s)

Relates https://github.com/gitpod-io/gitpod/issues/12248

## How to test

Run the test

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by utam0k** [comment]
I can get successful in four rows except for failing to prepare preview env.
- https://werft.gitpod-dev.com/job/gitpod-custom-to-fix-test.25
- https://werft.gitpod-dev.com/job/gitpod-custom-to-fix-test.26
- https://werft.gitpod-dev.com/job/gitpod-custom-to-fix-test.29
- https://werft.gitpod-dev.com/job/gitpod-custom-to-fix-test.30

File: test/tests/workspace/contexts_test.go [hunk 1]
```diff
@@ -126,30 +118,47 @@ func runContextTests(t *testing.T, tests []ContextTest) {
 				{Name: "pvc", FF: "persistent_volume_claim"},
 			}
 
+			for _, ff := range ffs {
+				func() {
```
Comment (jenting) [inline test/tests/workspace/contexts_test.go:122]: `func() { }` is redundant.

File: test/tests/components/ws-manager/prebuild_test.go [hunk 2]
```diff
@@ -109,7 +138,7 @@ func TestPrebuildWorkspaceTaskFail(t *testing.T) {
 				t.Fatalf("cannot start workspace: %q", err)
 			}
 
-			t.Cleanup(func() {
+			defer func() {
```
Comment (jenting) [inline test/tests/components/ws-manager/prebuild_test.go:141]: Perhaps you could summarize when to use defer and when to use t.Cleanup.

**Review by jenting — APPROVED** [review]
I will not block this PR and see what's the nightly test result.

**Pull request merged**